test/vni-manager-mocha.js

```diff
@@ -0,0 +1,306 @@
+/**
+ * File: vni-manager-mocha.js
+ * Unit tests for the vni APIs defined in src/vni-manager.js
+ */
+
+var chai = require('chai');
+var expect = chai.expect;
+var should = chai.should();
+
+var componentFactory = require('../src/noflo-component-factory');
+var createState = require('../src/create-state');
+var inputStates = require('../src/input-states');
+var promiseOutput = require('../src/promise-output');
+
+var test = require('./common-test');
+
+var vniManager = require('../src/vni-manager');
+
+describe('vni-manager', function() {
+
+    beforeEach(function() {
+
+        component = test.createComponent(componentFactory({
+            inPorts:{input:{
+                ondata: promiseOutput(function(payload){
+                    return payload + " world";
+                })
+            }},
+            outPorts: promiseOutput.outPorts
+          }, vniManager )
+        );
+
+    });
+
+    afterEach(function() {
+        // clean up
+        component.deleteAllVnis();
+    });
+
+    it('should exist as a function', function() {
+      vniManager.should.exist;
+      vniManager.should.be.a('function');
+    });
+
+    describe('#vnis', function() {
+        it('should exist as an object in the component instance', function() {
+            component.vnis.should.exist;
+            component.vnis.should.be.a('object');
+            Object.keys( component.vnis ).should.have.length( 0 );
+        });
+    });
+
+    describe('#vni', function() {
+
+        it('should create and return a new IIP vni if no vnid is specified and no vni exists', function() {
+
+            var testVni = component.vni();
+
+            testVni.should.be.an('object');
+            testVni.inputStates.should.exist;
+            testVni.inputStates.should.be.a('function');
+            testVni.errorState.should.exist;
+            testVni.errorState.should.be.a('function');
+            testVni.outputState.should.exist;
+            testVni.outputState.should.be.a('function');
+
+            expect( testVni.parentVni ).to.be.undefined;
+            expect( testVni.previousLms ).to.be.undefined;
```

Comment:
The previousLms would be a property of the output or error states -- not directly in the VNI